### ISSUE TITLE
Spin fails to generate Bucklescript project even when npm is installed

### ISSUE THREAD

**Issue by johnridesabike** [tmattio/spin#67]
Using Spin 0.5.1 on MacOS and running `spin new bs-react spin-test`

```
🏗️  Creating a new bs-react in spin-test
Done!
[...config options here]
🎁  Installing packages. This might take a couple minutes.
😱  External command not available: yarn. Please install it and try again.
```

I don't have yarn installed. It would be nice if was smart enough to use either yarn or npm.

**Comment by tmattio** [comment]
Hi @johnridesabike, thanks for reporting the issue!

Without adding extra complexity and detecting installed commands, a fix for this could be to use `npm` instead of `yarn`, since I think it comes by default when installing `node`, so it would be safer to assume that users have `npm` installed on their systems.

**Comment by johnridesabike** [comment]
Would it make sense to make an option for it? I’m imagining something like:

```
Install dependencies with npm? [Y/n]
```

And if you select `n` it can say something like:

```
Setup complete. Before you begin development, you’ll need to install the dependencies with npm or yarn.
```

I imagine that anyone with yarn installed is familiar enough with it to know how to run it themselves. This approach is still flexible enough to let users use whichever package manager they want.

**Comment by tmattio** [comment]
@johnridesabike I'm a bit hesitant to add new prompts. I feel like Spin already asks too many questions when generating projects. If there is a satisfying solution to this that does not require adding a new prompt, I'd prefer to look into it.

In fact, thinking about your first suggestion, I think the cost of the added complexity of using one command or the other depending on the user's system will be justified if the alternative is to add another prompt.

**Comment by citizen428** [comment]
@tmattio While not quite as convenient, skipping the post-install command and outputting a message like "Please run yarn/npm install to install all dependencies" seems like an option that could at least be considered.

**Comment by tmattio** [comment]
@citizen428 Good idea! Actually, we could tweak the behavior of the post-install commands to make them optional. If the post-install command is optional, it will run only if the user has the command on its system, otherwise, it would just ignore it, or eventually write a non-error message. What do you think?

EDIT: Maybe that's what you meant by "skipping the post-install"?

**Comment by citizen428** [comment]
Your proposed solution is slightly more sophisticated than what I had in mind, I wouldn't even have attempted to run the command, just unconditionally output the message. 

I like your approach though, as we'd essentially just have to change the following pattern match:

```reasonml
        switch (
          Utils.Sys.exec("which", ~args=[|el.command|]) |> Lwt_main.run
        ) {
        | WEXITED(0) => ()
        | _ => raise(Errors.External_command_unavailable(el.command))
        };
```

Instead of raising `Errors.External_command_unavailable` we could just print an error like 

```
${command} unvailable.
Please run ${command} ${args}" 
```

On second thought, there's a lot happening after that `switch`, which we may need to move into the first match clause.

**Comment by citizen428** [comment]
@tmattio I quickly hacked something together, this is what it looks like:

<img width="476" alt="Screen Shot 2020-03-26 at 23 24 58" src="https://user-images.githubusercontent.com/47985/77670663-6a683c80-6f7e-11ea-9dd0-798cc26134d9.png">

**Comment by johnridesabike** [comment]
For what it's worth, Create React App will automatically use either yarn or npm: https://github.com/facebook/create-react-app/blob/master/packages/create-react-app/createReactApp.js#L333

**Comment by citizen428** [comment]
Thanks @johnridesabike! But I do think it makes more sense for Spin to have consistent handling of all post-install commands rather than a special case for `npm`/`yarn`.

**Comment by tmattio** [comment]
A fix for this is implemented in #74. The `bs-react` template will use `yarn` if it is available, or `npm` if not.

We define this in the template like this:
```
(post_gen
  (actions 
    (run yarn install))
  (message "🎁  Installing packages. This might take a couple minutes.")
  (enabled_if (not (run which yarn))))

(post_gen
  (actions 
    (run npm install))
  (message "🎁  Installing packages. This might take a couple minutes.")
  (enabled_if (run which yarn)))
```

Closing this now since I want to merge the PR in the next days (and tidy up the issue board at the same time).
